From: José Roberto de Souza <jose.souza@intel.com>
Date: Mon, 22 Sep 2025 09:48:13 -0700
Subject: gallium/llvmpipe/test: Rename rsqrtf() to _rsqrtf()
Origin: upstream, commit:5a482a7a3d1973f9c86a22f31050e2d2acbc8450
Bug: https://gitlab.freedesktop.org/mesa/mesa/-/issues/13797
Last-Update: 2026-08-14

Some versions of math.h export rsqrtf(), causing Mesa's local test helper
with the same name to fail compilation. Rename the test helper.

--- a/src/gallium/drivers/llvmpipe/lp_test_arit.c
+++ b/src/gallium/drivers/llvmpipe/lp_test_arit.c
@@ -197,7 +197,8 @@ const float rcp_values[] = {
 };
 
 
-static float rsqrtf(float x)
+/* Some versions of math.h exports rsqrtf() while others don't. */
+static float _rsqrtf(float x)
 {
    return 1.0/(float)sqrt(x);
 }
@@ -323,7 +324,7 @@ unary_tests[] = {
    {"exp", &lp_build_exp, &expf, exp2_values, ARRAY_SIZE(exp2_values), 18.0 },
    {"log", &lp_build_log_safe, &logf, log2_values, ARRAY_SIZE(log2_values), 20.0 },
    {"rcp", &lp_build_rcp, &rcpf, rcp_values, ARRAY_SIZE(rcp_values), 20.0 },
-   {"rsqrt", &lp_build_rsqrt, &rsqrtf, rsqrt_values, ARRAY_SIZE(rsqrt_values), 20.0 },
+   {"rsqrt", &lp_build_rsqrt, &_rsqrtf, rsqrt_values, ARRAY_SIZE(rsqrt_values), 20.0 },
    {"sin", &lp_build_sin, &sinf, sincos_values, ARRAY_SIZE(sincos_values), 20.0 },
    {"cos", &lp_build_cos, &cosf, sincos_values, ARRAY_SIZE(sincos_values), 20.0 },
    {"sgn", &lp_build_sgn, &sgnf, sgn_values, ARRAY_SIZE(sgn_values), 20.0 },
